pkg/render/whisker/component.go

```diff
@@ -0,0 +1,318 @@
+	objs := []client.Object{
+		render.CreateNamespace(WhiskerNamespace, c.cfg.Installation.KubernetesProvider, render.PSSRestricted, c.cfg.Installation.Azure),
+	}
+
+	objs = append(objs,
+		render.CreateOperatorSecretsRoleBinding(WhiskerNamespace),
+		c.serviceAccount(),
+		c.clusterRole(),
+		c.clusterRoleBinding(),
+		c.goldmaneService(),
+		c.whiskerService(),
+		c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace))
+
+	objs = append(objs, c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace))
+	objs = append(objs, c.deployment())
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(WhiskerNamespace, c.cfg.PullSecrets...)...)...)
+
+	return objs, nil
+}
+
+func (c *Component) Ready() bool {
+	return true
+}
+
+func (c *Component) serviceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: WhiskerServiceAccountName, Namespace: WhiskerNamespace},
+	}
+}
+
+func (c *Component) whiskerContainer() corev1.Container {
+	return corev1.Container{
+		Name:            WhiskerContainerName,
+		Image:           c.whiskerImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		Env: []corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "INFO"},
+			{Name: "CA_CERT_PATH", Value: c.cfg.TrustedCertBundle.MountPath()},
+		},
+		SecurityContext: securitycontext.NewNonRootContext(),
+		VolumeMounts:    c.cfg.TrustedCertBundle.VolumeMounts(rmeta.OSTypeLinux),
+	}
+}
+
+func (c *Component) whiskerService() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "whisker",
+			Namespace: WhiskerNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{Port: 8081}},
+			Selector: map[string]string{
+				"k8s-app": WhiskerDeploymentName,
+			},
+		},
+	}
+}
+
+func (c *Component) whiskerBackendContainer() corev1.Container {
+	return corev1.Container{
+		Name:            WhiskerBackendContainerName,
+		Image:           c.whiskerBackendImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		Env: []corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "INFO"},
+			{Name: "CA_CERT_PATH", Value: c.cfg.TrustedCertBundle.MountPath()},
+			{Name: "PORT", Value: "3002"},
+			{Name: "GOLDMANE_HOST", Value: "localhost:7443"},
+		},
+		SecurityContext: securitycontext.NewNonRootContext(),
+		VolumeMounts:    c.cfg.TrustedCertBundle.VolumeMounts(rmeta.OSTypeLinux),
+	}
+}
+
+func (c *Component) goldmaneContainer() corev1.Container {
+	env := []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "INFO"},
+		{Name: "CA_CERT_PATH", Value: "/certs/tls.crt"},
+		{Name: "PORT", Value: "7443"},
+	}
+	var volumeMounts []corev1.VolumeMount
+	if c.cfg.LinseedPublicCASecret != nil {
+		env = append(env, corev1.EnvVar{
+			Name:  "PUSH_URL",
+			Value: "https://localhost:8080/api/v1/flows/bulk"})
+		volumeMounts = []corev1.VolumeMount{
+			secretMount("/certs", c.cfg.LinseedPublicCASecret),
```

Comment:
How does this secret get into the namespace? I don't see it rendered in the PR, unless I'm missing something.

Reply:
It's via the install manifest retrieved from CC.

Reply:
This is removed for now and I'll add it back when guardian is added.